lua/astrocommunity/pack/cpp/init.lua

```diff
@@ -46,6 +46,14 @@ return {
                 if assert(vim.lsp.get_client_by_id(args.data.client_id)).name == "clangd" then
                   require "clangd_extensions"
                   vim.api.nvim_del_augroup_by_name "clangd_extensions"
+                  require("astrocore").set_mappings({
+                    n = {
+                      ["<Leader>lw"] = {
+                        "<cmd>ClangdSwitchSourceHeader<CR>",
+                        desc = "Clangd: Switch source/header file",
+                      },
+                    },
+                  }, {})
                 end
               end,
             },
```

Comment:
```suggestion
            },
          },
          clangd_extension_mappings = {
            {
              event = "LspAttach",
              desc = "Load clangd_extensions with clangd",
              callback = function(args)
                if assert(vim.lsp.get_client_by_id(args.data.client_id)).name == "clangd" then
                  require("astrocore").set_mappings({
                    n = {
                      ["<Leader>lw"] = { "<Cmd>ClangdSwitchSourceHeader<CR>", desc = "Switch source/header file" },
                    },
                  }, { buffer = args.buf })
                end
              end,
            },
```

Create a new mapping that attaches with each `clangd` that doesn't delete itself and handles creating the buffer local mapping